master/buildbot/www/authz/endpointmatchers.py

```diff
@@ -174,17 +174,19 @@ def match_BuildEndpoint_rebuild(self, epobject, epdict, options):
         build = yield epobject.get({}, epdict)
         defer.returnValue(Match(self.master, build=build))
 
-#####
-# not yet implemented
-
 
 class ViewBuildsEndpointMatcher(EndpointMatcherBase):
 
-    def __init__(self, branch=None, project=None, builder=None, **kwargs):
+    def __init__(self, **kwargs):
         EndpointMatcherBase.__init__(self, **kwargs)
-        self.branch = branch
-        self.project = project
-        self.builder = builder
+
+    def match(self, ep, action="get", options=None):
+        if not "builds" in ep:
+            return None
+        return defer.succeed(Match(self.master))
```

Comment:
this will not restrict all build elements. Only those that you can access via "builds/xxx"
but you can also access logs via "/logs/:logid", which will contain probably the data you want to restrict. The logids are easy to guess as they are incremental